arrows/core/applets/dump_klv.cxx

```diff
@@ -110,86 +122,130 @@ run()
+    kv::config_block_formatter fmt( config );
     fmt.print( fout );
     std::cout << "Wrote config to \"" << out_file << "\". Exiting.\n";
     return EXIT_SUCCESS;
   }
 
-  if( !kwiver::vital::algo::video_input::check_nested_algo_configuration( "video_reader", config ) )
+  if( !kva::video_input::check_nested_algo_configuration( "video_reader", config ) )
   {
     std::cerr << "Invalid video_reader config" << std::endl;
     return EXIT_FAILURE;
   }
 
+  if ( !kva::serialize_metadata::check_nested_algo_configuration( "metadata_serializer",
+                                                                  config ) )
+  {
+    std::cerr << "Invalid metadata_serializer config" << std::endl;
+    return EXIT_FAILURE;
+  }
+
   // instantiate a video reader
   try
   {
     video_reader->open( video_file );
   }
-  catch ( kwiver::vital::video_exception const& e )
+  catch ( kv::video_exception const& e )
   {
     std::cerr << "Video Exception-Couldn't open \"" << video_file << "\"" << std::endl
               << e.what() << std::endl;
     return EXIT_FAILURE;
   }
-  catch ( kwiver::vital::file_not_found_exception const& e )
+  catch ( kv::file_not_found_exception const& e )
   {
     std::cerr << "Couldn't open \"" << video_file << "\"" << std::endl
               << e.what() << std::endl;
     return EXIT_FAILURE;
   }
 
-  kwiver::vital::algorithm_capabilities const& caps = video_reader->get_implementation_capabilities();
-  if ( ! caps.capability( kwiver::vital::algo::video_input::HAS_METADATA ) )
+  kv::algorithm_capabilities const& caps = video_reader->get_implementation_capabilities();
+  if ( ! caps.capability( kva::video_input::HAS_METADATA ) )
   {
     std::cerr << "No metadata stream found in " << video_file << '\n';
     return EXIT_FAILURE;
   }
 
   int count(1);
-  kwiver::vital::image_container_sptr frame;
-  kwiver::vital::timestamp ts;
-  kwiver::vital::wrap_text_block wtb;
+  kv::image_container_sptr frame;
+  kv::timestamp ts;
+  kv::wrap_text_block wtb;
+  kv::metadata_map::map_metadata_t frame_metadata;
+
   wtb.set_indent_string( "    " );
 
+  // Avoid repeated dictionary access
+  bool detail = cmd_args["detail"].as<bool>();
+  bool quiet = cmd_args["quiet"].as<bool>();
+  bool log = cmd_args.count("log");
+
   while ( video_reader->next_frame( ts ) )
   {
-    std::cout << "========== Read frame " << ts.get_frame()
-              << " (index " << count << ") ==========" << std::endl;
+    if ( !quiet )
+    {
+      std::cout << "========== Read frame " << ts.get_frame()
+                << " (index " << count << ") ==========" << std::endl;
+    }
+
+    kv::metadata_vector metadata = video_reader->frame_metadata();
 
-    kwiver::vital::metadata_vector metadata = video_reader->frame_metadata();
-    for( auto meta : metadata )
+    if ( log )
     {
-      std::cout << "\n\n---------------- Metadata from: " << meta->timestamp() << std::endl;
+      int timestamp = ts.get_frame();
```

Comment:
FYI, from the "reasons I like `auto`" front... `int` is the wrong type here. (That said, the temporary really isn't needed at all...)